File: temps/flutter/lib/components/auto/mockPointer.dart
import '../../common/FN.dart';
import '../../common/mixin.dart';
import './index.dart';

playRecord(ME, List arr) async {
  var cursor = $Global.value['previewCursor'];
  Map $cursor = cursor.value;

  double ox = $cursor['x'];
  double oy = $cursor['y'];

  cursor.value['useTransition'] = false;

  for (var obj in arr) {
    if (obj == 'E') {
      PS.publish('ProxyMouseupSync');

      return;
    }

    int x = obj['x'];
    int y = obj['y'];
    int dx = obj['dx'];
    int dy = obj['dy'];

    ME.x = x.toDouble();
    ME.y = y.toDouble();
    ME.dx = dx.toDouble();
    ME.dy = dy.toDouble();

    cursor.value['x'] = ox + dx * unit;
    cursor.value['y'] = oy + dy * unit;

    await FN.sleep(34);
  }
}

class AutoMouseEvent {
  double x = 0;
  double y = 0;
  double dx = 0;
  double dy = 0;
}

playMouseRecord(eid) {
  AutoMouseEvent ME = AutoMouseEvent();

  List arr = ($Global.value['interactionRecord'])[eid];

  playRecord(ME, arr);

  return ME;
}

cleanMouseRecord() {
  $Global.value['interactionRecord'] = {};
  $Global.value['previewEventMap'] = {};
  $Global.value['previewCursor'].value['x'] = -20.0;
  $Global.value['previewCursor'].value['y'] = -20.0;
}

File: temps/Web-Vue3/src/lib/auto/mockPointer.ts
import FN from '../../common/FN'
import GV from '../GV/index'
import { $store } from '../../store/index'
const Global = $store.history

async function playRecord(ME, arr) {
  let cursor = Global.previewCursor
  let ox = cursor.x
  let oy = cursor.y

  cursor.useTransition = false

  for (let obj of arr) {
    if (obj == 'E') {
      FN.PS.publish('ProxyMouseupSync')

      return
    }

    let { x, y, dx, dy } = obj

    ME.x = x
    ME.y = y
    ME.dx = dx
    ME.dy = dy
    cursor.x = ox + dx
    cursor.y = oy + dy

    await GV.sleep(34)
  }
}

export function playMouseRecord(eid) {
  let ME = {
    x: 0, y: 0, dx: 0, dy: 0
  }

  let arr = Global.interactionRecord[eid]

  playRecord(ME, arr)

  return ME
}

export function cleanMouseRecord() {
  Global.interactionRecord = {}
  Global.previewEventMap = {}
  Global.previewCursor.x = -20
  Global.previewCursor.y = -20
}
